src/mlpack/methods/ann/layer/inception_layer.hpp

```diff
@@ -0,0 +1,290 @@
+      convPool(inMaps, poolProj, 1, 1, 1, 1, 1, 1),
+      pool3(3),
+      bias1(out1, bias),
+      biasProj3(projSize3, bias),
+      bias3(out3, bias),
+      biasProj5(projSize5, bias),
+      bias5(out5, bias),
+      biasPool(poolProj, bias)
+  {
+    /* Nothing to do. */
+  }
+
+  // perform forward passes for all the layers.
+
+  template<typename eT>
+  void Forward(const arma::Cube<eT>& input, arma::Cube<eT>& output)
+  {
+    conv1.InputParameter() = input;
+
+    // Forward pass for 1x1 conv path.
+    conv1.Forward(conv1.InputParameter(), conv1.OutputParameter());
+    bias1.Forward(conv1.OutputParameter(), bias1.OutputParameter());
+    base1.InputParameter() = bias1.OutputParameter();
+    base1.Forward(bias1.OutputParameter(), base1.OutputParameter());
+
+    // Forward pass for 3x3 conv path.
+    proj3.InputParameter() = input;
+    proj3.Forward(input, proj3.OutputParameter());
+    biasProj3.Forward(proj3.OutputParameter(), biasProj3.OutputParameter());
+    baseProj3.InputParameter() = biasProj3.OutputParameter();
+    baseProj3.Forward(biasProj3.OutputParameter(), baseProj3.OutputParameter());
+    conv3.InputParameter() = baseProj3.OutputParameter();
+    conv3.Forward(baseProj3.OutputParameter(), conv3.OutputParameter());
+    bias3.Forward(conv3.OutputParameter(), bias3.OutputParameter());
+    base3.InputParameter() = bias3.OutputParameter();
+    base3.Forward(bias3.OutputParameter(), base3.OutputParameter());    
+    
+    // Forward pass for 5x5 conv path.
+    proj5.InputParameter() = input;
+    proj5.Forward(input, proj5.OutputParameter());
+    biasProj5.Forward(proj5.OutputParameter(), biasProj5.OutputParameter());
+    baseProj5.InputParameter() = biasProj5.OutputParameter();
+    baseProj5.Forward(biasProj5.OutputParameter(), baseProj5.OutputParameter());
+    conv5.InputParameter() = baseProj5.OutputParameter();
+    conv5.Forward(baseProj5.OutputParameter(), conv5.OutputParameter());
+    bias5.Forward(conv5.OutputParameter(), bias5.OutputParameter());
+    base5.InputParameter() = bias5.OutputParameter();
+    base5.Forward(bias5.OutputParameter(), base5.OutputParameter());
+    
+    // Forward pass for 3x3 pool path. 
+    pool3.InputParameter() = input;
+    pool3.Forward(input, pool3.OutputParameter());
+    Pad(pool3.OutputParameter(), 1, 1, convPool.InputParameter());
+    convPool.Forward(pool3.OutputParameter(), convPool.OutputParameter());
+    biasPool.Forward(convPool.OutputParameter(), biasPool.OutputParameter());
+    basePool.InputParameter() = biasPool.OutputParameter();
+    basePool.Forward(convPool.OutputParameter(), basePool.OutputParameter());
+
+    // concatenate outputs of all the paths.
+    output = arma::join_slices( 
+              arma::join_slices(
+                arma::join_slices( 
+                  base1.OutputParameter(), base3.OutputParameter() ), 
+                  base5.OutputParameter() ), basePool.OutputParameter());
+
```

Comment:
As you already pointed out the inception layer is basically a container for a couple of convolution and pooling layer. Since they are abstracted through the inception layer, the main network has now knowledge about the containing layer. So, the inception layer has to forward any weights to any capsulated layer.